app/queries/get_content_collection.rb

```diff
@@ -87,5 +86,9 @@ def live_version(item)
         @live_versions[item.id]
       end
     end
+
+    def select_output_fields_only(hash)
+      hash.present.as_json.select { |k, v| output_fields.map(&:to_s).include?(k) }
+    end
```

Comment:
You can probably use [Hash#slice](http://api.rubyonrails.org/classes/Hash.html#method-i-slice) for this. You won't need to `#to_s` the fields if you slice before calling `#as_json`.